experimental/unitsconversion/tests/units_test.rs

```diff
@@ -0,0 +1,155 @@
+    use core::str::FromStr;
+    use icu_unitsconversion::measureunit::MeasureUnitParser;
+    use num::BigRational;
+    use std::collections::HashSet;
+
+    /// Convert a decimal number to a BigRational.
+    fn convert_decimal_to_rational(decimal: &str) -> Option<BigRational> {
+        let mut components = decimal.split('.');
+        let integer_component = components.next().unwrap_or("0");
+        let decimal_component = components.next().unwrap_or("0");
+        let decimal_length = decimal_component.chars().count() as i32;
+
+        if components.next().is_some() {
+            return None;
+        }
+
+        let mut integer_component = BigRational::from_str(integer_component).unwrap();
+        let decimal_component = BigRational::from_str(decimal_component).unwrap();
+
+        let ten = BigRational::from_integer(10.into());
+        let decimal_component = decimal_component / ten.pow(decimal_length);
+        integer_component += decimal_component;
+        Some(integer_component)
+    }
+
+    /// Convert a scientific notation string to a BigRational.
+    pub fn get_rational(rational: &str) -> Option<BigRational> {
+        // remove all the commas
+        let rational = rational.replace(',', "");
+
+        let mut parts = rational.split('E');
+        let rational_part = parts.next().unwrap_or("1");
+        let exponent_part = parts.next().unwrap_or("0");
+        if parts.next().is_some() {
+            return None;
+        }
+
+        let mut rational_part = convert_decimal_to_rational(rational_part)?;
+        let exponent_part = i32::from_str(exponent_part).unwrap();
+
+        let ten = BigRational::from_integer(10.into());
+        let exponent_part = ten.pow(exponent_part);
+        rational_part *= exponent_part;
+        Some(rational_part)
+    }
+
+    #[test]
+    fn test_conversion() {
+        // let provider = icu_unitsconversion::provider::UnitsInfoV1;
+
+        /// Represents a test case for units conversion.
+        #[derive(Debug)]
+        struct UnitsTest {
+            _category: String,
+            input_unit: String,
+            output_unit: String,
+            _result: BigRational,
+        }
+
+        let data = std::fs::read_to_string("tests/data/unitsTest.txt").unwrap();
+        let tests: Vec<UnitsTest> = data
+            .lines()
+            .filter(|line| !line.starts_with('#') && !line.is_empty())
+            .map(|line| {
+                let parts: Vec<&str> = line.split(';').map(|s| s.trim()).collect();
+                UnitsTest {
+                    _category: parts[0].to_string(),
+                    input_unit: parts[1].to_string(),
+                    output_unit: parts[2].to_string(),
+                    _result: get_rational(parts[4]).unwrap(),
+                }
+            })
+            .collect();
+
+        let parser = MeasureUnitParser::from_payload(
+            &icu_unitsconversion::provider::Baked::SINGLETON_UNITS_INFO_V1.units_conversion_trie,
```

Comment:
are you following the transliteration model with baked data in the `tests/` folder, or are you properly hooking up baked date with a `icu_units_conversion_data` crate? In the former case, this is not correct, in the latter you don't need the files in `tests/data`

Reply:
As I saw at the transliteration, they have copied the baked data into tests.

Can I read the data directly from the Datagen place?

Reply:
Actually, we can just access the data directly.